lib/circuit.js

```diff
@@ -423,18 +443,27 @@ function fallback (circuit, err, args) {
 }
 
 function fail (circuit, err, args, latency) {
-  /**
-   * Emitted when the circuit breaker action fails
-   * @event CircuitBreaker#failure
-   */
-  circuit.emit('failure', err, latency);
-
-  // check stats to see if the circuit should be opened
-  const stats = circuit.stats;
-  const errorRate = stats.failures / stats.fires * 100;
-  if (errorRate > circuit.options.errorThresholdPercentage ||
-    circuit.options.maxFailures >= stats.failures) {
-    circuit.open();
+  if (circuit[FAILURE_CHECK_FUNCTION] &&
+    !(circuit[FAILURE_CHECK_FUNCTION].call(circuit[FAILURE_CHECK_FUNCTION], err))) {
```

Comment:
As a general note, I'm not sure I like this naming convention. It makes the code a bit awkward by negating the `if` condition. Something more like `circuit[ERROR_OK]` maybe? That would make this code read a little better perhaps:

```js
if (circuit[ERROR_OK] && circuit[ERROR_OK].call(circuit[ERROR_OK], err)) {
  circuit.emit('success', err, latency);
}
```

Perhaps the class method could be `circuit.errorOkFilter( func )`.